### DISCUSSION TITLE
Page Grouping in pages folder

### DISCUSSION THREAD

**Discussion by pijusz** [withastro/docs#3273] [Ideas]
I've production pages that already have predetermined format, but the amount of .astro files in single folder is starting to become annoying.
As Next.js introduced with /app dir, there are (group) folders, that allow you to group pages in file structure.

Pages folder structure:
```
pages/
├── index.astro
└── (group)/
    ├── item1.astro
    └── item2.astro
```
Result:
```
index.html
item1.html
item2.html
```

There will definitely be some collision issues with pages. Where there are multiple pages with same name. It should fail during build.

**Comment by doctorlocke** [comment]
Coming from Next and Remix this is something I would really miss not having. Once a project starts getting large with a lot of pages, being able to group things like login, register... under a hidden directory like (auth) becomes a very nice feature.

**Comment by ArmandPhilippot** [comment]
Thanks for opening this. I know this is has been a while but because this is not something we can do in docs (we only document the current state of the API), I think it is time to close this discussion.
If you're still interested, there is roadmap discussion which seems to be for the same feature https://github.com/withastro/roadmap/discussions/480 and I encourage you to join the discussion/upvote there instead of here.
